web-common/src/features/dashboards/state-managers/selectors/dimensions.ts

```diff
@@ -27,7 +27,13 @@ export const visibleDimensions = ({
   validExplore,
   dashboard,
 }: DashboardDataSources): MetricsViewSpecDimension[] => {
-  if (!validMetricsView?.dimensions || !validExplore?.dimensions) return [];
+  if (
+    !validMetricsView?.dimensions ||
+    !validExplore?.dimensions ||
+    !dashboard?.visibleDimensions
```

Comment:
This might work, but I believe the `visibleDimensions` should always be defined. So the error might be upstream where we initially create the `MetricsExplorerEntity`?